### PR TITLE
Minor improvements to `cf_OpenLibrary()`’s HOG2 id detection code

### PR DESCRIPTION
**Pull request by Jayman2000**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [ ] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [x] Other changes

### Description

This PR contains two minor improvements to the code that detects whether or not `.hog` files actually begin with the ASCII characters HOG2. See the commit messages for details.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [ ] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

**Comment by GravisZro** [comment]
It would be better to simply replace `#define HOG_TAG_STR "HOG2"` with a std::string_view. However, my understanding is that  @winterheart plans on replacing `cfile.*` and `hogfile.*` entirely which would negate any work on them.

File: cfile/hogfile.h [hunk 1]
```diff
@@ -62,9 +62,11 @@
 
 #include <cstdint>
 #include <cstdio>
+#include <string>
 
 #define HOG_HDR_SIZE (64)
 #define HOG_TAG_STR "HOG2"
+#define HOG_TAG_LEN std::string::traits_type::length(HOG_TAG_STR)
```
Comment (winterheart) [inline cfile/hogfile.h:69]: No need to call std::string::length, we already knows that there always be 4 bytes. Just define it as is:

```cpp
#define HOG_TAG_LEN (4)
```

**Review by winterheart — COMMENTED** [review]
I'd prefer move header check into `ReadHogHeader()` function of hogfile.cpp (with some refactoring and additional sane checks). There a lot of low-level checks in `cf_OpenLibrary()` already.

File: cfile/hogfile.h (hunk 1 above)
Comment (Jayman2000) [reply cfile/hogfile.h]: OK, done.

File: cfile/hogfile.h [hunk 2]
```diff
@@ -62,9 +62,11 @@
 
 #include <cstdint>
 #include <cstdio>
+#include <string>
```
Comment (winterheart) [inline cfile/hogfile.h:65]: This include unneeded.

Comment (Jayman2000) [reply cfile/hogfile.h]: Good catch. I just force pushed a new version that doesn’t have that line.

**Comment by winterheart** [comment]
OK, ready to merge, thank you!

**Pull request merged**